PasswordValidator_UnitTests/AdvancedValidator_UnitTests.cs

```diff
@@ -10,7 +12,14 @@ internal class AdvancedValidator_UnitTests
 
         public AdvancedValidator_UnitTests()
         {
-            _validator = new AdvancedValidator();
+            var rules = new List<IRule>
```

Comment:
All this has changed now - you will use 'mock' rules - it no longer matter what rules you actually put into it - might be a clue here as to where this is going..